sdk/eidolon_ai_sdk/system/agent_controller.py

```diff
@@ -271,12 +273,11 @@ async def agent_event_stream(self, handler, process, last_state, **kwargs) -> As
         is_async_gen = inspect.isasyncgenfunction(handler.fn)
         stream = handler.fn(self.agent, **kwargs) if is_async_gen else self.stream_agent_fn(handler, **kwargs)
 
-        def get_start_event():
+        async def get_start_event():
```

Comment:
do we need to make this extra mongo call? haven't we just retrieved the process doc?